### DISCUSSION TITLE
Any plans on supporting Unix Domain Sockets?

### DISCUSSION THREAD

**Discussion by Code-Codex** [ni/grpc-device#470] [Ideas]
Are there any plans on supporting unix domain sockets. These are available for both windows and linux and will assit with IPC on the same machine.

[AB#1762814](https://ni.visualstudio.com/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/1762814)

**Comment by harsha-bhushan** [comment]
@Code-Codex 
Thanks for contacting. Is there a specific reason why your application would specifically require UDS? We do currently support gRPC connection over TCP loopback localhost.

**Comment by Code-Codex** [comment]
@harsha-bhushan 
Background:
Our application is made up of multiple processes on the same machine. Today we directly interface with Dac-MX libraries to communicate with devices.
Our next generation product is based on .NET 6 which is not a supported .NET version for NI libraries, so we have been looking at this project to assist with migration.

Few reasons for UDS:
1. Performance - On the same machine gRPC over UDS allows to bypass the network stack allowing for faster communication and lower latencies.
2. UAC requirements - UDS doesnt require an application to run in a privileged mode. There is no UAC prompt.

Hope this helps.

**Comment by harsha-bhushan** [comment]
@Code-Codex 
Thanks for the reply. I'm the product owner at NI R&D responsible for gRPC. At the moment we have not prioritized UDS. But we are always willing to adjust our roadmap based on user input & feedback. Would you be interested in a call to discuss further? You can reach me at harsha.bhushan@ni.com. I look forward to hearing from you!

**Comment by harsha-bhushan** [comment]
Classifying this as a feature request and not a bug & hence converting this to a discussion.
